code/datums/components/material/material_container.dm

```diff
@@ -75,7 +78,6 @@
 	RegisterSignal(atom_target, COMSIG_ATOM_REQUESTING_CONTEXT_FROM_ITEM, PROC_REF(on_requesting_context_from_item))
 
 /datum/component/material_container/Destroy(force)
-	retrieve_all()
```

Comment:
Technically we still want to dump all when the component and NOT the atom attached is being destroyed.

Reply:
There is no use case for that currently so unless required we can leave here. Currently only machines, rcd uses a material container and both of them retain their material container till the end of their lifetime, no deleting the component in between